### PR TITLE
Add MJ and GJ

### PR DESCRIPTION
**Pull request by SgtBatten**
## Proposed change

Add MJ and GJ as part of https://github.com/home-assistant/architecture/discussions/859

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/86055
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by emontnemery** [comment]
Please fix the conflict and update source/_integrations/number.markdown too 👍

File: source/_integrations/sensor.markdown [hunk 1]
```diff
@@ -32,7 +32,7 @@ The type of data a sensor returns impacts how it is displayed in the frontend. T
 - **date**: Date string (ISO 8601)
 - **distance**: Generic distance in km, m, cm, mm, mi, yd, or in
 - **duration**: Duration in days, hours, minutes or seconds
-- **energy**: Energy in Wh, kWh or MWh
+- **energy**: Energy in Wh, kWh, MWh, MJ or GJ
```
Comment (epenet) [inline source/_integrations/sensor.markdown]: please also adjust to `number.markdown`

File: source/_integrations/number.markdown [hunk 2]
```diff
@@ -33,7 +33,7 @@ The type of data a number represents impacts how it is displayed in the frontend
 - **data_rate**: Data rate in bit/s, kbit/s, Mbit/s, Gbit/s, B/s, kB/s, MB/s, GB/s, KiB/s, MiB/s, or GiB/s
 - **data_size**: Data size in bit, kbit, Mbit, Gbit, B, kB, MB, GB, TB, PB, EB, ZB, YB, KiB, MiB, GiB, TiB, PiB, EiB, ZiB, or YiB
 - **distance**: Generic distance in km, m, cm, mm, mi, yd, or in
-- **energy**: Energy in Wh, kWh or MWh
+- **energy**: Energy in Wh, kWh, MWh, MJ or GJ
 - **frequency**: Frequency in Hz, kHz, MHz or GHz
```
Comment (epenet) [inline source/_integrations/number.markdown]:
```suggestion
- **energy**: Energy in Wh, kWh, MWh, MJ, or GJ
- **frequency**: Frequency in Hz, kHz, MHz, or GHz
```

File: source/_integrations/sensor.markdown [hunk 3]
```diff
@@ -32,7 +32,7 @@ The type of data a sensor returns impacts how it is displayed in the frontend. T
...
 - **enum**: Has a limited set of (non-numeric) states
 - **frequency**: Frequency in Hz, kHz, MHz or GHz
```
Comment (epenet) [inline source/_integrations/sensor.markdown]:
```suggestion
- **energy**: Energy in Wh, kWh, MWh, MJ, or GJ
- **enum**: Has a limited set of (non-numeric) states
- **frequency**: Frequency in Hz, kHz, MHz, or GHz
```

**Review by epenet — COMMENTED** [review]
It's not enforced everywhere (yet) but could you please add a comma before `or`?

File: source/_integrations/number.markdown (hunk 2 above)
Comment (SgtBatten) [reply source/_integrations/number.markdown]: Oh sorry i went the wrong way then with the conflict, ill fix

**Review by epenet — APPROVED** [review]
LGTM
This should included in 2023.2 (not quite sure how cherry-picking/milestones work here)

**Comment by SgtBatten** [comment]
Awesome, thanks for the assistance.

**Review by frenck — APPROVED** [review]
Thanks, @SgtBatten 👍 

../Frenck

**Pull request merged**